frame/nfts/src/functions.rs

```diff
@@ -54,6 +54,16 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Item::<T, I>::get(&collection, &item).ok_or(Error::<T, I>::UnknownItem)?;
 		with_details(&collection_details, &mut details)?;
 
+		if details.deposit.account == details.owner {
+			// Move the deposit to the new owner.
+			T::Currency::repatriate_reserved(
```

Comment:
Do you have a test that the reserved balance carries over?

Reply:
yep, we do:
```
		...
		assert_ok!(Nfts::transfer(RuntimeOrigin::signed(1), 0, 42, 2));
		assert_eq!(Balances::reserved_balance(&1), 0);
		assert_eq!(Balances::reserved_balance(&2), 1);
```